vault/provider.go

```diff
@@ -102,6 +103,10 @@ func Provider() *schema.Provider {
 								Type: schema.TypeString,
 							},
 						},
+						"iam": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
```

Comment:
I agree that it would be nicer if this was more like the CLI, but beyond `method=aws`, there aren't any uses for this and could ultimately confuse users. Do you think something like `signAWSRequest=true` or something might be better?

Reply:
I was thinking a non-boolean option would leave open the possibility for other auth method handling in the future. Like I notice gcp auth has a similar token signing function in the vault CLI: https://github.com/hashicorp/vault/blob/170521481d58961761bfd6877221b005cf0f4c7e/vendor/github.com/hashicorp/vault-plugin-auth-gcp/plugin/cli.go#L95-L95
I'm not overly familiar with how all these fit together, but it just strikes me as the sort of thing we may need to add to TFVP in the future.